cloud-agent/service/server/src/main/scala/org/hyperledger/identus/agent/server/jobs/IssueBackgroundJobs.scala

```diff
@@ -2,40 +2,68 @@ package org.hyperledger.identus.agent.server.jobs
 
 import org.hyperledger.identus.agent.server.config.AppConfig
 import org.hyperledger.identus.agent.server.jobs.BackgroundJobError.ErrorResponseReceivedFromPeerAgent
-import org.hyperledger.identus.agent.walletapi.model.error.DIDSecretStorageError.WalletNotFoundError
+import org.hyperledger.identus.agent.walletapi.service.ManagedDIDService
+import org.hyperledger.identus.agent.walletapi.storage.DIDNonSecretStorage
 import org.hyperledger.identus.castor.core.model.did.*
 import org.hyperledger.identus.mercury.*
 import org.hyperledger.identus.mercury.protocol.issuecredential.*
 import org.hyperledger.identus.pollux.core.model.*
 import org.hyperledger.identus.pollux.core.model.error.CredentialServiceError
 import org.hyperledger.identus.pollux.core.service.CredentialService
+import org.hyperledger.identus.resolvers.DIDResolver
+import org.hyperledger.identus.shared.messaging
+import org.hyperledger.identus.shared.messaging.{Message, WalletIdAndRecordId}
+import org.hyperledger.identus.shared.messaging.MessagingService.RetryStep
+import org.hyperledger.identus.shared.models.{WalletAccessContext, WalletId}
 import org.hyperledger.identus.shared.models.Failure
 import org.hyperledger.identus.shared.utils.aspects.CustomMetricsAspect
 import org.hyperledger.identus.shared.utils.DurationOps.toMetricsSeconds
 import zio.*
 import zio.metrics.*
 
+import java.util.UUID
+
 object IssueBackgroundJobs extends BackgroundJobsHelper {
 
-  val issueCredentialDidCommExchanges = {
-    for {
+  private val TOPIC_NAME = "issue"
+
+  val issueFlowsHandler = messaging.MessagingService.consumeWithRetryStrategy(
+    "identus-cloud-agent",
+    IssueBackgroundJobs.handleMessage,
+    Seq(
+      RetryStep(TOPIC_NAME, 5, 0.seconds, s"$TOPIC_NAME-retry-1"),
```

Comment:
Should we try to implement the following pattern:
[topic -> retry_topic -> dead_letters_queue](https://forum.confluent.io/t/how-should-you-implement-a-retry-delay-queue-using-kafka-topics/3588)

Reply:
Retrying on the same topic is identified as "flawed" by the author => _"... If the retry failed it would produce the event back onto the topic. The main reason this was flawed was because when the delay was observed, it blocked the Consumer, this would also block consumption of other events, increasing the observed delay, essentially meaning events would be blocked for much longer than the expected time...."_